elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/timegrains/Day.java

```diff
@@ -34,14 +33,13 @@ public Day deserialize(Object val) {
 
             try {
                 if (val instanceof String) {
-                    date = new Day(new Timestamp(FORMATTER.parse((String) val).getTime()));
-                } else {
-                    date = new Day(FORMATTER.parse(FORMATTER.format(val)));
+                    date = new Day(TimeGrainFormatter.formatDateString(FORMATTER, (String) val));
                 }
+                date = new Day(FORMATTER.parse(FORMATTER.format(val)));
```

Comment:
Don't we still need the else case?  Otherwise this line stomps on line 36.

Reply:
This line will catch ISO formatted date object/ dateString. It will ensure the output is always 'yyyy-MM-dd' format.